ui/hooks/accounts/useMultichainClient.ts

```diff
@@ -1,32 +1,54 @@
 import { KeyringClient, Sender } from '@metamask/keyring-api';
 import { HandlerType } from '@metamask/snaps-utils';
 import { CaipChainId, Json, JsonRpcRequest } from '@metamask/utils';
+import { SnapId } from '@metamask/snaps-sdk';
 import { useMemo } from 'react';
 import {
   handleSnapRequest,
   multichainUpdateBalance,
 } from '../../store/actions';
 import { BITCOIN_WALLET_SNAP_ID } from '../../../shared/lib/accounts/bitcoin-wallet-snap';
+import { SOLANA_WALLET_SNAP_ID } from '../../../shared/lib/accounts/solana-wallet-snap';
+
+export enum WalletClientType {
+  Bitcoin = 'bitcoin-wallet-snap',
+  Solana = 'solana-wallet-snap',
+}
+
+const SNAP_ID_MAP: Record<WalletClientType, SnapId> = {
+  [WalletClientType.Bitcoin]: BITCOIN_WALLET_SNAP_ID,
+  [WalletClientType.Solana]: SOLANA_WALLET_SNAP_ID,
+};
+
+export class MultichainWalletSnapSender implements Sender {
+  private snapId: SnapId;
+
+  constructor(snapId: SnapId) {
+    this.snapId = snapId;
+  }
 
-export class BitcoinWalletSnapSender implements Sender {
   send = async (request: JsonRpcRequest): Promise<Json> => {
     // We assume the caller of this module is aware of this. If we try to use this module
     // without having the pre-installed Snap, this will likely throw an error in
     // the `handleSnapRequest` action.
     return (await handleSnapRequest({
       origin: 'metamask',
-      snapId: BITCOIN_WALLET_SNAP_ID,
+      snapId: this.snapId,
       handler: HandlerType.OnKeyringRequest,
       request,
     })) as Json;
   };
 }
 
-export class BitcoinWalletSnapClient {
+export class MultichainClient {
```

Comment:
Similar to the `MultichainWalletSnapSender`, I think this one should be `MultichainWalletSnapClient`.

The hook (file) should also be renamed `useMultichainWalletSnapClient` too.

WDYT?